service/src/test/java/bio/terra/workspace/service/workspace/flight/aws/CreateDeleteAwsContextFlightTest.java

```diff
@@ -0,0 +1,238 @@
+                    jobService.getStairway(),
+                    DeleteCloudContextFlight.class,
+                    deleteInputs,
+                    STAIRWAY_FLIGHT_TIMEOUT,
+                    /*debugInfo*/ null);
+
+    assertEquals(FlightStatus.SUCCESS, flightState.getFlightStatus());
+    assertTrue(awsCloudContextService.getAwsCloudContext(workspaceUuid).isEmpty());
+  }
+
+  @Test
+  public void undoCreateDeleteContextWithSecurityGroups() throws Exception {
+    AuthenticatedUserRequest userRequest = userAccessUtils.defaultUserAuthRequest();
+    UUID workspaceUuid = createWorkspace();
+    assertTrue(awsCloudContextService.getAwsCloudContext(workspaceUuid).isEmpty());
+
+    // Cause a failure on the set egress/ingress step, as undo for this step is a no-op
+    Map<String, StepStatus> doErrorStep =
+            Map.of(SetWorkspaceApplicationEgressIngressStep.class.getName(), StepStatus.STEP_RESULT_FAILURE_FATAL);
+
+    FlightDebugInfo debugInfo =
+            FlightDebugInfo.newBuilder()
+                    .doStepFailures(doErrorStep)
+                    .build();
+
+    FlightMap createInputs =
+            WorkspaceFixtures.createCloudContextInputs(
+                    workspaceUuid, userRequest, CloudPlatform.AWS, spendUtils.defaultGcpSpendProfile());
+
+    FlightState flightState =
+            StairwayTestUtils.blockUntilFlightCompletes(
+                    jobService.getStairway(),
+                    CreateCloudContextFlight.class,
+                    createInputs,
+                    STAIRWAY_FLIGHT_TIMEOUT,
+                    debugInfo);
+
+    assertEquals(FlightStatus.ERROR, flightState.getFlightStatus());
+    assertTrue(awsCloudContextService.getAwsCloudContext(workspaceUuid).isEmpty());
+  }
+
+  private SpendProfileId getSpendProfile() {
+    return spendUtils.defaultSpendId();
+  }
+
+  private UUID createWorkspace() {
+    Workspace request =
+        WorkspaceFixtures.defaultWorkspaceBuilder(UUID.randomUUID())
+            .spendProfileId(getSpendProfile())
+            .build();
+    return workspaceService.createWorkspace(
+        request, null, null, null, userAccessUtils.defaultUserAuthRequest());
+  }
+
+  private Map<String, StepStatus> getCreateStepNameToStepStatusMap() {
+    // Retry steps once to validate idempotency.
+    Map<String, StepStatus> retrySteps = new HashMap<>();
+    retrySteps.put(
+        CreateWorkspaceApplicationSecurityGroupsStep.class.getName(),
+        StepStatus.STEP_RESULT_FAILURE_RETRY);
+    retrySteps.put(
+        SetWorkspaceApplicationEgressIngressStep.class.getName(),
+        StepStatus.STEP_RESULT_FAILURE_RETRY);
+    retrySteps.put(MakeAwsCloudContextStep.class.getName(), StepStatus.STEP_RESULT_FAILURE_RETRY);
+    retrySteps.put(
+        CreateCloudContextFinishStep.class.getName(), StepStatus.STEP_RESULT_FAILURE_RETRY);
+    return retrySteps;
+  }
+
+  private Map<String, StepStatus> getDeleteStepNameToStepStatusMap() {
```

Comment:
nit static 

also there's Map.of(key, value)

Reply:
Streamlined with Map.of(), thanks!